### PR TITLE
Upgrade snakeYaml and woodstox versions

### PR DESCRIPTION
**Pull request by rdulmina**
## Purpose
$title

Fixes: #40060

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/build.gradle [hunk 1]
```diff
@@ -130,10 +130,8 @@ configurations {
...
     testCompile.exclude group: 'org.slf4j', module: 'slf4j-simple'
     testCompile.exclude group: 'org.ops4j.pax.logging', module: 'pax-logging-api'
-    testCompile.exclude group: 'org.codehaus.woodstox', module: 'woodstox-core-asl'
-    testCompile.exclude group: 'org.codehaus.woodstox', module: 'stax2-api'
 }
 
 configurations.all {
     resolutionStrategy.preferProjectModules()
-}
+}
```
Comment (warunalakshitha) [inline tests/jballerina-unit-test/build.gradle:137]: missing new line at the end

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLParserNegativeTest.java [hunk 2]
```diff
@@ -36,22 +36,23 @@ public void testXmlArg(String xmlValue, String expectedErrorMessage) {
...
-                {"<!-- comments cannot have -- in it -->" , "[1,29]\nMessage: The string \"--\" is not permitted " +
-                        "within comments."},
-                {"<-note>irshad</-note>", "[1,2]\nMessage: The markup in the document preceding the root element " +
-                        "must be well-formed."},
-                {"<note><.b>irshad</.b></note>", "[1,8]\nMessage: The content of elements must consist of " +
-                        "well-formed character data or markup."},
-                {"<?pi cannot contain ?> in it?><book></book>", "[1,24]\nMessage: Content is not allowed in prolog."}
+                {"<book name=\"irshad<\"></book>" , "Unexpected '<'  in attribute value\n" +
+                        " at [row,col {unknown-source}]: [1,19]"},
+                {"<!-- comments cannot have -- in it -->" , "[com.ctc.wstx.exc.WstxLazyException] String '--' not " +
```
Comment (warunalakshitha) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLParserNegativeTest.java:48]: Lets create an issue to remove com.ctc.wstx.exc.WstxLazyException from message

**Comment by warunalakshitha** [comment]
We need to test whether this will change the error messages and if changes this will be a breaking change as well. In that case better to include in the release note.

**Comment by rdulmina** [comment]
> We need to test whether this will change the error messages and if changes this will be a breaking change as well. In that case better to include in the release note.

Compared error messages with 1.2.37, there is no change in error messages

**Review by warunalakshitha — APPROVED** [review]

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLParserNegativeTest.java (hunk 2 above)
Comment (rdulmina) [reply tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/xml/XMLParserNegativeTest.java:48]: Created https://github.com/ballerina-platform/ballerina-lang/issues/40127

**Pull request merged**